src/Microsoft.DocAsCode.MarkdigEngine.Extensions/TripleColon/CodeExtension.cs

```diff
@@ -0,0 +1,258 @@
+        {
+            if(!string.IsNullOrEmpty(range) && !string.IsNullOrEmpty(id))
+            {
+                logError($"You must set only either Range or Id, but not both.");
+            }
+            
+            var codeSections = new List<string>();
+
+            if (!string.IsNullOrEmpty(range))
+            {
+                codeSections = GetCodeSectionsFromRange(range, code, codeSections, logError);
+            }
+            else if (!string.IsNullOrEmpty(id))
+            {
+                var strRegexStart = @"((<|</|#region)\s*" + id + @"(>|(\s*>)))";
+                var idRegexStart = new Regex(strRegexStart, RegexOptions.IgnoreCase);
+                var idRegexEnd = new Regex(strRegexStart + @"|#endregion", RegexOptions.IgnoreCase);
+                var codeLines = code.Split('\n').ToList();
+                var beg = codeLines.FindIndex(oo => idRegexStart.IsMatch(oo)) + 2;
+                var end = codeLines.FindLastIndex(oo => idRegexEnd.IsMatch(oo));
+                codeSections = GetCodeSectionsFromRange($"{beg}-{end}", code, codeSections, logError);
+            }
+            else
+            {
+                codeSections.Add(code);
+            }
+
+            if (codeSections == null)
+            {
+                return string.Empty;
+            }
+
+            codeSections = Dedent(codeSections);
+            var source = string.Join("    ...\n", codeSections.ToArray());
+            return source;
+        }
+
+        public static List<string> Dedent(List<string> sections)
+        {
+            var indentRegex = new Regex(@"^\s*\W");
+            var RemoveIndentSpacesRegexString = @"^[ \t]{{1,{0}}}";
+            var dedentedSections = new List<string>();
+            foreach (var section in sections)
+            {
+                var codeLines = section.Split('\n');
+                var length = codeLines.Where(oo => !string.IsNullOrWhiteSpace(oo))
+                                      .Min(oo =>
+                {
+                    var matches = indentRegex.Matches(oo);
+                    if (matches.Count > 0)
+                    {
+                        var match = matches[0];
+                        return match.Value.Length;
+                    } else
+                    {
+                        return 0;
+                    }
+                });
+                var normalizedLines = (length == 0 ? codeLines : codeLines.Select(s => Regex.Replace(s, string.Format(RemoveIndentSpacesRegexString, length), string.Empty))).ToArray();
+                dedentedSections.Add(string.Join("\n", normalizedLines));
+            }
+            return dedentedSections;
+        }
+
+        private static List<string> GetCodeSectionsFromRange(string range, string code, List<string> codeSections, Action<string> logError)
+        {
+            var codeLines = code.Split('\n');
+
+            range = range.Replace(" ", "");
+            var ranges = range.Split(',');
```

Comment:
with `RemoveEmptyEntries`, line 209 can be removed?
```suggestion
            var ranges = range.Split(new[] { "," }, StringSplitOptions.RemoveEmptyEntries)
```

Reply:
It turns out that with int.TryParse, it's totally unnecessary to remove spaces :). I just needed to remove 209.